### ISSUE TITLE
setting_a_maintenance_window in Redis

### ISSUE THREAD

**Issue by pratik-anand** [hashicorp/terraform-provider-google#9622]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment. If the issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If the issue is assigned to a user, that user is claiming responsibility for the issue. If the issue is assigned to "hashibot", a community member has claimed the issue already.

### Description

A new feature came in GCP  Redis for `setting a maintenance window`, which we need to set up.

### What happens if I don’t set a maintenance window?
If you don’t set up a maintenance window, your instance(s) can be updated at any time within the planned maintenance rollout period.
Setting a preferred maintenance window will not prevent the expected service disruption due to maintenance, but it will allow you to select a time frame that better suits your project(s) needs.

![image](https://user-images.githubusercontent.com/67919486/126746472-e29049cc-00cf-4f67-979f-c74725d34e3f.png)

### New or Affected Resource(s)

* google_redis_instance

### Potential Terraform Configuration

```tf
# Propose what you think the configuration to take advantage of this feature should look like.
# We may not use it verbatim, but it's helpful in understanding your intent.
```

### References

* https://cloud.google.com/memorystore/docs/redis/managing-maintenance-updates#setting_a_maintenance_window

**Comment by Sytten** [comment]
Bumping the thread since this is kind of critical for a lot of businesses.

**Comment by philipp-hb** [comment]
any news?

**Comment by jeroenj** [comment]
This is a duplicate of https://github.com/hashicorp/terraform-provider-google/issues/9727 which has been released as part of version [4.9.0](https://github.com/hashicorp/terraform-provider-google/releases/tag/v4.9.0): https://github.com/hashicorp/terraform-provider-google/pull/10978.

**Comment by melinath** [comment]
Closing since this is resolved.
